media-video/obs-studio/obs-studio-28.1.1.ebuild

```diff
@@ -0,0 +1,278 @@
+	truetype? (
+		media-libs/fontconfig
+		media-libs/freetype
+	)
+	v4l? (
+		media-libs/libv4l
+		virtual/udev
+	)
+	vlc? ( media-video/vlc:= )
+	wayland? (
+		dev-libs/wayland
+		x11-libs/libxkbcommon
+	)
+"
+RDEPEND="${DEPEND}"
+
+QA_PREBUILT="
+	usr/lib*/obs-plugins/chrome-sandbox
+	usr/lib*/obs-plugins/libcef.so
+	usr/lib*/obs-plugins/libEGL.so
+	usr/lib*/obs-plugins/libGLESv2.so
+	usr/lib*/obs-plugins/libvk_swiftshader.so
+	usr/lib*/obs-plugins/libvulkan.so.1
+	usr/lib*/obs-plugins/swiftshader/libEGL.so
+	usr/lib*/obs-plugins/swiftshader/libGLESv2.so
+"
+
+pkg_setup() {
+	use lua && lua-single_pkg_setup
+	use python && python-single-r1_pkg_setup
+}
+
+src_unpack() {
+	default
+
+	if [[ ${PV} == 9999 ]]; then
+		local EGIT_SUBMODULES=()
+		use browser && EGIT_SUBMODULES+=( plugins/obs-browser )
+		use websocket && EGIT_SUBMODULES+=(
+			plugins/obs-websocket
+			plugins/obs-websocket/deps/asio
+			plugins/obs-websocket/deps/json
+			plugins/obs-websocket/deps/qr
+			plugins/obs-websocket/deps/websocketpp
+		)
+		git-r3_src_unpack
+	else
+		if use browser; then
+			rm -d ${P}/plugins/obs-browser || die
+			mv obs-browser-${OBS_BROWSER_COMMIT} ${P}/plugins/obs-browser || die
+		fi
+
+		if use websocket; then
+			rm -d ${P}/plugins/obs-websocket || die
+			mv obs-websocket-${OBS_WEBSOCKET_COMMIT} ${P}/plugins/obs-websocket || die
+
+			rm -d ${P}/plugins/obs-websocket/deps/asio || die
+			mv asio-${ASIO_COMMIT} ${P}/plugins/obs-websocket/deps/asio || die
+	￼
+			rm -d ${P}/plugins/obs-websocket/deps/json || die
+			mv json-${JSON_COMMIT} ${P}/plugins/obs-websocket/deps/json || die
+	￼
+			rm -d ${P}/plugins/obs-websocket/deps/qr || die
+			mv QR-Code-generator-${QR_COMMIT} ${P}/plugins/obs-websocket/deps/qr || die
+	￼
+			rm -d ${P}/plugins/obs-websocket/deps/websocketpp || die
+			mv websocketpp-${WEBSOCKETPP_COMMIT} ${P}/plugins/obs-websocket/deps/websocketpp || die
+		fi
+	fi
+}
+
+src_prepare() {
+	# Avoid errors if we don't have 'obs-browser' or 'obs-websocket'.
+	if ! use browser; then
+		sed -i 's/FATAL_ERROR "obs-browser/STATUS "obs-browser/' plugins/CMakeLists.txt || die
+	fi
+
+	if ! use websocket; then
+		sed -i 's/FATAL_ERROR "obs-websocket/STATUS "obs-websocket/' plugins/CMakeLists.txt || die
+	fi
+
+	cmake_src_prepare
+}
+
+src_configure() {
+	local libdir=$(get_libdir)
+	local mycmakeargs=(
+		$(usev browser -DCEF_ROOT_DIR=../${CEF_DIR})
+		-DENABLE_ALSA=$(usex alsa)
+		-DENABLE_AJA=OFF
+		-DENABLE_BROWSER=$(usex browser)
+		-DENABLE_DECKLINK=$(usex decklink)
+		-DENABLE_FREETYPE=$(usex truetype)
+		-DENABLE_JACK=$(usex jack)
+		-DENABLE_LIBFDK=$(usex fdk)
+		-DENABLE_NEW_MPEGTS_OUTPUT=OFF # Requires librist and libsrt.
+		-DENABLE_PIPEWIRE=$(usex pipewire)
+		-DENABLE_PULSEAUDIO=$(usex pulseaudio)
+		-DENABLE_RTMPS=$(usex ssl ON OFF)
+		-DENABLE_SPEEXDSP=$(usex speex)
+		-DENABLE_V4L2=$(usex v4l)
+		-DENABLE_VLC=$(usex vlc)
+		-DENABLE_VST=ON
+		-DENABLE_WAYLAND=$(usex wayland)
+		-DENABLE_WEBSOCKET=$(usex websocket)
```

Comment:
Yeah, this is pretty ugly.  They annoyingly have the flags to disable the submodules... in the submodules... and I figured they're bit big to clone if not used (a bit over 300 MiB, json being the biggest at around 200).

That said, yeah, it would be a lot cleaner to just let them in.

Reply:
For live ebuild I wouldn't worry too much, the .tar.gz is just 6.6MB (still big'ish but well).

It's also possible to use `EGIT_CLONE_TYPE="shallow"` in make.conf to save a bit on cloning (does occasionally cause issues with build systems that inspect their own git history though, some ebuilds need to override that).

Reply:
Of course there's also the route to try and use `dev-cpp/nlohmann_json`, albeit can be a pain sometimes.

Reply:
Right now I went with just always getting them, but those are interesting ideas.  If not now, could try them out in the future.